addons/base_vat/models/res_partner.py

```diff
@@ -306,6 +307,27 @@ def check_vat_al(self, vat):
             return True
         return False
 
+    @api.model
+    def check_vat_do(self, vat):
+        """
+        Validate Dominican Republic tax numbers:
+        - RNC (9 digits, modulus-11 via stdnum.do.rnc)
+        - Cédula (11 digits, Luhn check)
+        """
+        number = ''.join(ch for ch in vat if ch.isdigit())
+
+        try:
+            rnc.validate(number)
+            return True
+        except (InvalidLength, ValidationError, InvalidChecksum):
+            pass
+
+        try:
+            luhn.validate(number)
+            return True
+        except (ValidationError, InvalidChecksum):
+            return False
```

Comment:
You can prevent validating the VAT twice by checking the length.

```suggestion
        try:
            if len(number) == 9:
                rnc.validate(number)
            elif len(number) == 11:
                luhn.validate(number)
            else:
                return False
            return True
        except (ValidationError, InvalidChecksum):
            return False
```

Reply:
makes a lot of sense! Thanks :)